src/components/InlineEditable/index.js

```diff
@@ -48,87 +48,83 @@ const Editable = styled('span')`
 	${props => props.css};
 `;
 
-class InlineEditable extends Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-			isEditing: props.isEditing || false,
-			value: props.value,
-		};
-	}
-
-	handleFocus = () => {
-		if (this.props.disabled) return;
+function InlineEditable({
+	isEditing: isEditingProps,
+	value: valueProps,
+	disabled,
+	onFocusOut,
+	type,
+	placeholder,
+	className,
+	innerRef,
+	nameCss,
+	editableCss,
+	placeholderCss,
+}) {
+	const [isEditing, setIsEditing] = useState(isEditingProps || false);
+	const [value, setValue] = useState(valueProps);
+
+	const handleFocus = useCallback(() => {
+		if (disabled) return;
 
 		let shouldBeCleared = false;
 
-		if (this.state.isEditing) {
-			shouldBeCleared = this.props.onFocusOut(this.state.value);
-		}
-		this.setState(state => ({
-			value: shouldBeCleared ? '' : state.value,
-			isEditing: !this.state.isEditing,
-		}));
-	};
-
-	handleChange = (e) => {
-		this.setState({
-			value: e.target.value,
-		});
-	};
-
-	render() {
-		const {isEditing, value} = this.state;
-		const {
-			type, placeholder, className, innerRef, disabled,
-		} = this.props;
-
 		if (isEditing) {
-			return (
-				<NameInput
-					ref={innerRef}
-					type={type}
-					value={value}
-					onChange={this.handleChange}
-					onBlur={this.handleFocus}
-					placeholder={placeholder}
-					autoFocus
-					flexible
-					className={className}
-					css={this.props.nameCss}
-					onKeyPress={(e) => {
-						if (e.key === 'Enter') {
-							e.target.blur();
-						}
-					}}
-				/>
-			);
+			shouldBeCleared = onFocusOut(value);
 		}
+		setIsEditing(!isEditing);
+		setValue(shouldBeCleared ? '' : value);
+	}, [isEditing, value, disabled, onFocusOut]);
 
-		if (value) {
-			return (
-				<Editable
-					className={className}
-					onClick={this.handleFocus}
-					disabled={disabled}
-					css={this.props.editableCss}
-				>
-					{value}
-				</Editable>
-			);
-		}
+	const handleChange = useRef((e) => {
```

Comment:
I think useCallback is more appropriate in that case

Reply:
I don't think so, useCallback will always have to check for a reference change There is no possible reference change here. This never changes so it is basically a ref.

Reply:
we could also use useMemo(() => () => {}) to avoid the current prop.